js/index.js

```diff
@@ -101,3 +101,96 @@ modalTriggers.forEach((trigger) => {
     });
   });
 });
+
+function popup(data) {
+  if (
+    data.incompleteUserDetails === false &&
+    data.roles.developer === undefined &&
+    data.roles.designer === undefined &&
+    data.roles.maven === undefined &&
+    data.roles.productmanager === undefined
+  ) {
+    const popupContainer = document.querySelector('.roles-container');
+    const submitButton = document.querySelector('.role-button button');
+    const checkboxes = document.querySelectorAll('.checkbox-input');
+    const spinner = submitButton.querySelector('.spinner');
+    const roles = {};
+
+    const registerUserRoles = async (roles) => {
+      spinner.style.display = 'inline-block';
+      submitButton.style.backgroundColor = '#ccc';
+      const updateRoles = {
+        roles: {
+          ...data.roles,
+          ...roles,
+        },
+      };
+      const res = await fetch('https://api.realdevsquad.com/users/self', {
+        method: 'PATCH',
+        body: JSON.stringify(updateRoles),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+      });
+      return res;
+    };
+
+    function showPopup() {
+      popupContainer.style.display = 'flex';
+    }
+
+    async function hidePopup() {
+      const response = await registerUserRoles(roles);
+      if (response.status === 204) {
+        spinner.style.display = 'none';
+        submitButton.style.backgroundColor = '#008000';
+        submitButton.disabled = false;
+        popupContainer.style.display = 'none';
+      } else {
+        //have to add toast here
+        //will work on next pr
```

Comment:
Why on next PR?

Reply:
Done with changes. can you please check latest commit message